packages/core/database/lib/entity-manager/relationsOrderer.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const _ = require('lodash/fp');
+
+// TODO: Document this class properly
+/* Constraints:
+  - Expects you will never connect a relation before / after one that does not exist
+  - Expect the last initArr element to be the last relation in the database
+*/
+class FractionalOrderer {
+  constructor(initArr, idColumn, orderColumn) {
+    this.arr = _.castArray(initArr || []).map((r) => ({
+      init: true,
+      id: r[idColumn],
+      order: r[orderColumn],
+    }));
+  }
+
+  _updateRelationOrder(r) {
+    let idx;
+    if (r.position.before) {
+      const { idx: _idx, relation } = this.findRelation(r.position.before);
+      if (relation.init) r.order = relation.order - 0.5;
+      else r.order = relation.order;
+      idx = _idx;
+    } else if (r.position.after) {
+      const { idx: _idx, relation } = this.findRelation(r.position.after);
+      if (relation.init) r.order = relation.order + 0.5;
+      else r.order = relation.order;
+      idx = _idx + 1;
+    } else if (r.position.start) {
+      r.order = 0.5;
+      idx = 0;
+    } else {
+      const lastRelation = this.arr.at(-1);
+      // TODO: Use a big number instead of lastRelation.order
+      if (lastRelation.init) r.order = lastRelation.order + 0.5;
+      else r.order = lastRelation.order;
+      idx = this.arr.length;
+    }
+
+    return { relation: r, idx };
+  }
+
+  // TODO: Improve performance by using a map
+  findRelation(id) {
+    const idx = this.arr.findIndex((r) => r.id === id);
```

Comment:
could you just use `.find` here? iirc it breaks the loop once it's found the relation you're after.

Reply:
I think findIndex behaves the same. Could you elaborate on why you find it more appropriate to use the .find?
Also, I need the idx value of the relation in the `connect` method below.

Reply:
ah sorry, I missed that you used `idx`. This makes sense then 😄